soc/renesas/ra/ra4m1/soc.c

```diff
@@ -134,3 +134,5 @@ const struct opt_set_mem ops __attribute__((section(".opt_set_mem"))) = {
 	}
 };
 #endif
+
+const unsigned short __weak function_irq_table[CONFIG_SW_ISR_TABLE_MAX_FUNCTION_NUMBER] = { };
```

Comment:
Sorry, but aside from the fact that this table is HUGE for the RA family, most of it other than 32 items (limitation of RA ICU) will never be used. I would suggest to keep only used items, but not any possible.

Reply:
I would prefer to spend memory to my application, instead of wasting it. 

Btw the RA2 ICU is a different creature. It's incompatible with this PR.